zio-crypto/shared/src/main/scala/zio/crypto/hashing/Hashing.scala

```diff
@@ -0,0 +1,115 @@
+package zio.crypto.hashing
+
+import zio._
+import zio.crypto.ByteHelpers
+
+import java.nio.charset.Charset
+import java.security.{ MessageDigest => JMessageDigest }
+
+sealed class HashAlgorithm(val name: String)
+
+object HashAlgorithm {
+  case object SHA256 extends HashAlgorithm("SHA-256")
+  case object SHA512 extends HashAlgorithm("SHA-512")
+
+  // DO NOT USE IN A SECURE CONTEXT
+  case object MD5 extends HashAlgorithm("MD5")
+  // DO NOT USE IN A SECURE CONTEXT
+  case object SHA1 extends HashAlgorithm("SHA-1")
+}
+
+case class MessageDigest[T](value: T) extends AnyVal
+
+object Hashing {
+
+  type Hashing = Has[Hashing.Service]
+
+  trait Service {
+    def hash(m: Array[Byte], alg: HashAlgorithm): Task[MessageDigest[Array[Byte]]]
```

Comment:
`Chunk[Byte]` should always be used instead of `Array[Byte]`. `Array` is mutable, but `Chunk` is not, and Chunk does not box bytes or other primitives, like Array (it's based on Array).